targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F437xG/TARGET_UBLOX_C030/min_battery_voltage.cpp

```diff
@@ -17,11 +17,11 @@
 #include "min_battery_voltage.h"
 #include "battery_charger_i2c.h"
 
-/** Defining HAL_MspInit strong function
+/** Defining mbed_main strong function
  * in user defined file as described in documentation
  */
 
-void HAL_MspInit(void)
+void mbed_main(void)
```

Comment:
This is `mbed_main: User's code to be executed before main().` . Should not be defined in the targets

What problem is here? Why a target code needs kernel to be initialized ?

Reply:
See #5995: our C030 board requires configuration of a voltage level in its battery charger chip if it is to work reliably.  This battery charger chip is only configurable over I2C.  The [standard ST] I2C drivers make calls to `wait_us()`, which is an RTOS function.  So our target code needs to do this configuration call _after_ the RTOS has been initialised and _before_ entry to `main()`.  Is there such a hook, other than `mbed_main()`?

Reply:
> Is there such a hook, other than mbed_main()?

No, there is not.

target code should use what it has, target code, in this case us ticker functionality.

Reply:
So I guess it is a rule that target code should not use RTOS calls.  Is that written down somewhere, just so that I can have the discussion with ST (whose code it is)?

Reply:
Grepping around a little, there are calls to `wait_ms()` and `wait_us()` all over the place in the target code so I don't see how it can be a rule that target code should not use RTOS calls.  The other option would be to introduce a hook for target configuration to occur after the RTOS has been initialised, since I can see that being generally useful.  I will raise a separate issue for this.